tests/test_grading.py

```diff
@@ -224,85 +237,50 @@ def check_visible_scores(user, assignment, hidden=(), visible=()):
+        self.assertTrue("Saved published scores for {}".format(self.assignment.display_name) in source)
         for user in users:
             check_visible_scores(user, self.assignment, hidden=['Total', 'Composition'])
             check_visible_scores(user, self.assignment2, hidden=['Total'], visible=['Composition'])
 
         self.login(self.staff1.email)
         endpoint = '/admin/course/{}/assignments/{}/publish'.format(self.course.id, self.assignment2.id)
-        response = self.client.post(endpoint, data={}, follow_redirects=True)
-        self.assert_200(response)
-        for user in users:
-            check_visible_scores(user, self.assignment, hidden=['Total', 'Composition'])
-            check_visible_scores(user, self.assignment2, visible=['Composition', 'Total'])
-
-        # Cannot publish an already published grade
-        self.login(self.staff1.email)
-        endpoint = '/admin/course/{}/assignments/{}/publish'.format(self.course.id, self.assignment2.id)
-        response = self.client.post(endpoint, data={}, follow_redirects=True)
-        self.assert_200(response)
-        source = response.get_data().decode('utf-8')
-        self.assertTrue("{} scores for {} already published".format('Total', self.assignment2.display_name) in source)
-        for user in users:
-            check_visible_scores(user, self.assignment, hidden=['Total', 'Composition'])
-            check_visible_scores(user, self.assignment2, visible=['Composition', 'Total'])
-
-        # Cannot hide a nonpublished grade
-        self.login(self.staff1.email)
-        endpoint = '/admin/course/{}/assignments/{}/publish'.format(self.course.id, self.assignment.id)
-        response = self.client.post(endpoint, data={"hide":True}, follow_redirects=True)
-        self.assert_200(response)
-        source = response.get_data().decode('utf-8')
-        self.assertTrue("{} scores for {} already hidden".format('Total', self.assignment.display_name) in source)
+        response = publish_scores(self.assignment2, ['Composition', 'Total'])
         for user in users:
             check_visible_scores(user, self.assignment, hidden=['Total', 'Composition'])
             check_visible_scores(user, self.assignment2, visible=['Composition', 'Total'])
 
         # If assignment is not visible, still can publish
         self.assignment.visible = False
         self.login(self.staff1.email)
-        endpoint = '/admin/course/{}/assignments/{}/publish'.format(self.course.id, self.assignment.id)
-        response = self.client.post(endpoint, data={}, follow_redirects=True)
-        self.assert_200(response)
+        response = publish_scores(self.assignment, ['Total'])
         source = response.get_data().decode('utf-8')
-        self.assertTrue("Published {} {} scores".format(self.assignment.display_name, 'Total') in source)
+        self.assertTrue("Saved published scores for {}".format(self.assignment.display_name) in source)
```

Comment:
Sidenote: I wish we could have used the `assertMessageFlashed` flask-testing gives use